### PR TITLE
Get agents information outside collect agents method

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#12795|

## Description

This PR moves the method in charge to get the agents' information from the Global database as well as the OS information from an agent outside the method that collects the active agents. Since as part of this epic, the vulnerability detector will scan a specific agent from the task in the task manager we need to get the information for active agents for a scan by interval, and for a unique agent in case of a scan on demand.

## Scan-build report

image

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Comment by MiguelazoDS** [comment]
# Dod

## Environment 

image

## Scan types

### Baseline scan

- Log
image
- No alerts
image

### Partial scan

Deleting and installing some packages in agents to get alerts.
- Log
image
- Alerts
image
image

### Full scan

Completely remove the vulnerabilities inventory to get all alerts.
- Log
image
- Alerts
image

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Just a couple of comments. Good change!

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -5175,357 +5186,379 @@ void *wm_vuldet_main(wm_vuldet_t * vuldet) {
...
-        }
-        else {
-            id = id_array[i++];
-        }
-
-        // Getting agent-info data from global.db
-        j_agent_info = wdb_get_agent_info(id, &sock);
-        if (!j_agent_info) {
-            mdebug1("Failed to get agent '%d' information from Wazuh DB.", id);
+int wm_vuldet_collect_agent_info(int agent_id, wm_vuldet_t *vuldet, scan_agent *agents, int *sock) {
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: Changed!

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -7691,10 +7722,9 @@ void wm_vuldet_run_scan(wm_vuldet_t *vuldet) {
         return;
     }
 
-    if (wm_vuldet_collect_agents_to_scan(vuldet)) {
+    if (wm_vuldet_collect_agents_to_scan(vuldet, TRUE)) {
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:7727]: Added!

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**